triggers/simulations/overview.rst

```diff
@@ -0,0 +1,461 @@
+      val (client, actAs) = Await.result(setup, simulationConfig.simulationSetupTimeout)
+      // Ledger process (as an Akka typed actor) being defined
+      val ledger: ActorRef[LedgerProcess.Message] =
+        context.spawn(LedgerProcess.create(client), "ledger")
+
+      // Trigger and external components could be defined here
+
+      Behaviors.empty
+    }
+  }
+
+Simulating External Ledger Interactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As external components may interact with a ledger (e.g. by creating or archiving contracts that a trigger registers an interest in), it is often necessary to model these within a given trigger simulation. This may be done by defining an Akka typed actor with type ``Behavior[Unit]`` and having this actor send ``LedgerProcess.ExternalAction`` messages to the ledger actor.
+
+For example, to model an external component that randomly creates instances of a ``Cat`` contract once every second, we could write:
+
+.. code-block:: scala
+  def createRandomCat(
+    ledger: ActorRef[LedgerProcess.Message],
+    actAs: Party
+  )(implicit materializer: Materializer): Behavior[Unit] =
+    Behaviors.withTimers[Unit] { timer =>
+      timer.startTimerWithFixedDelay((), 1.second, 1.second)
+
+      Behaviors.receiveMessage { _ =>
+        val randomCat =
+          CreatedEvent(
+            templateId = Some("Cats:Cat"),
+            createArguments = Some(SValue.SRecord(
+              "Cats:Cat",
+              fields = ImmArray("owner", "name"),
+              values = ArrayList(
+                SValue.SParty(actAs),
+                SValue.SInt64(Random.nextLong()),
+              ),
+            )),
+        )
+        val createEvent = LedgerProcess.ExternalAction(CreateContract(randomCat, actAs))
+
+        ledger ! createEvent
+        Behaviors.same
+      }
+    }
+
+where a ``Cat`` template might be defined as:
+
+.. code-block:: none
+  template Cat
+    with
+      owner : Party
+      name : Int
+    where
+      signatory owner
+
+.. note::
+  Currently, it is not possible to model external components that exercise choices on a contract. This is a known limitation.
+
+  Currently, a good of understanding of the low level Speedy machine ``SValue`` is required when defining create or archive events. This is a known limitation.
+
+Trigger Process Component
+-------------------------
+
+A primitive trigger may be thought of as having state machine like behaviour defined by the following Daml code:
+
+- its user defined Daml state - ``state`` say
+- its ``updateState`` Daml function
+
+  - crudely, we may think of this as defining a side effecting function with type ``TriggerMsg => state => state`` (side effects here being queries against the internal trigger ACS)
+- and its ``rule`` Daml function
+
+  - crudely, we may think of this as defining a side effecting function with type ``TriggerMsg => Party => state => Unit`` (side effects here being ledger command submissions).
+
```

Comment:
```suggestion
```

Reply:
Is there a particular reason you're suggesting this change?